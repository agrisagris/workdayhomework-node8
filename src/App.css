.container button {
  padding: 10px 20px;
  border-radius: 10px;
  margin-top: 10px;
  border-style: none;
  background-color: #f0f4f8;
}
.container button:hover {
  background-color: #d9e2ec;
}

.modal-content, .container button {
  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
}

.container {
  padding: 20px;
}
.container div {
  font-weight: bold;
}
.modal-overlay {
  background-color: rgba(0, 0, 0, 0.15);
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.modal-content {
  margin: auto;
  margin-top: 25vh;
  width: 200px;
  padding: 20px;
  background-color: white;
  border-radius: 6px;
}

table {
  margin-top: 20px;
  width: 300px;
}
table th {
  background: #d9e2ec;
  padding: 2px 10px;
}
table td {
  font-family: monospace;
  padding: 2px 10px;
}
table td:nth-child(even) {
  text-align: right;
}
table td:nth-child(odd) {
  text-align: center;
}
table tr:nth-child(even) {
  background: #f0f4f8;
}

/*# sourceMappingURL=App.css.map */
