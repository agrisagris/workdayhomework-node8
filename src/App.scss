%button {
  padding: 10px 20px;
  border-radius: 10px;
  margin-top: 10px;
  border-style: none;
  background-color: #f0f4f8;
  @extend %shadow;
  &:hover {
    background-color: #d9e2ec;
  }
}

%shadow {
  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1),
    0 2px 4px -1px rgba(0, 0, 0, 0.06);
}

.container {
  padding: 20px;
  div {
    font-weight: bold;
  }
  button {
    @extend %button;
  }
}

.modal-overlay {
  background-color: rgba(0, 0, 0, 0.15);
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.modal-content {
  margin: auto;
  margin-top: 25vh;
  width: 200px;
  padding: 20px;
  background-color: white;
  border-radius: 6px;
  @extend %shadow;
}

table {
  margin-top: 20px;
  width: 300px;
  th {
    background: #d9e2ec;
    padding: 2px 10px;
  }
  td {
    font-family: monospace;
    padding: 2px 10px;
  }
  td:nth-child(even) {
    text-align: right;
  }
  td:nth-child(odd) {
    text-align: center;
  }
  tr:nth-child(even) {
    background: #f0f4f8;
  }
}
