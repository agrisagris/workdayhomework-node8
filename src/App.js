import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import * as ratesActions from "./redux/actions/ratesActions";
import ratesSelector from "./redux/selectors/ratesSelector";
import RatesTable from "./components/RatesTable";
import "./App.css";

export default function App() {
  const dispatch = useDispatch();
  const rates = useSelector(ratesSelector);
  const [showDialog, setShowDialog] = useState(false);

  const toogleShowDialog = () => {
    setShowDialog(!showDialog);
  };
  const onLoadRates = () => {
    dispatch(ratesActions.loadRates());

    toogleShowDialog();
  };
  return (
    <div className="container">
      <div>Here be dragons</div>
      <button type="button" onClick={onLoadRates}>
        Load Rates
      </button>
      <RatesTable
        rates={rates}
        showDialog={showDialog}
        toogleShowDialog={toogleShowDialog}
      />
    </div>
  );
}
