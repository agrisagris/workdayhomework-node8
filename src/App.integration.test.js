import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import { Provider as ReduxProvider } from "react-redux";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";

import App from "./App";
import configureStore from "./redux/configureStore";

const component = () =>
  render(
    <ReduxProvider store={configureStore()}>
      <App />
    </ReduxProvider>
  );
var mock = new MockAdapter(axios);
const data = { rates: { EUR: 1, USD: 1.2264, JPY: 133.49 } };

test("Load rates button: pressed show data", async () => {
  mock.onGet(process.env.REACT_APP_API).reply(200, data);

  const { getByRole, getByText, container } = component();

  const buttonElement = getByRole("button", { name: /load rates/i });

  expect(buttonElement).toBeInTheDocument();
  fireEvent.click(buttonElement);
  await wait(() => {
    expect(mock.history.get.length).toBe(1);
  });

  expect(container).toMatchSnapshot();

  expect(getByText(/currency/i)).toBeInTheDocument();

  expect(getByText(/eur/i)).toBeInTheDocument();
  expect(getByText(/usd/i)).toBeInTheDocument();
  expect(getByText(/jpy/i)).toBeInTheDocument();
});

test("Load rates button: pressed network error", async () => {
  mock.reset();
  mock.onGet(process.env.REACT_APP_API).networkError();

  const { getByRole, container } = component();

  const buttonElement = getByRole("button", { name: /load rates/i });

  expect(buttonElement).toBeInTheDocument();
  fireEvent.click(buttonElement);

  await wait(() => {
    expect(mock.history.get.length).toBe(1);
  });

  expect(container).toMatchSnapshot();
});

test("Load rates button: pressed loading", async () => {
  mock.reset();
  mock.onGet(process.env.REACT_APP_API).replyOnce();

  const { getByRole, getByText, container } = component();

  const buttonElement = getByRole("button", { name: /load rates/i });

  expect(buttonElement).toBeInTheDocument();
  fireEvent.click(buttonElement);

  expect(getByText(/loading/i)).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
