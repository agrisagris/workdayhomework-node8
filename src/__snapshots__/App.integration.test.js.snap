// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load rates button: pressed loading 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      Here be dragons
    </div>
    <button
      type="button"
    >
      Load Rates
    </button>
    <div>
      Loading
    </div>
  </div>
</div>
`;

exports[`Load rates button: pressed network error 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      Here be dragons
    </div>
    <button
      type="button"
    >
      Load Rates
    </button>
    <div
      class="modal-overlay"
    >
      <div
        class="modal-content"
      >
        <p>
          Error retrieving data
        </p>
        <button
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Load rates button: pressed show data 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      Here be dragons
    </div>
    <button
      type="button"
    >
      Load Rates
    </button>
    <table>
      <thead>
        <tr>
          <th>
            Currency
          </th>
          <th>
            Rate
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            EUR
          </td>
          <td>
            1.00000
          </td>
        </tr>
        <tr>
          <td>
            USD
          </td>
          <td>
            1.22640
          </td>
        </tr>
        <tr>
          <td>
            JPY
          </td>
          <td>
            133.49000
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
